corelib/calc/crayfish_dataset_utils.cpp

```diff
@@ -130,6 +131,61 @@ void CrayfishDataSetUtils::populateSpatialFilter(DataSet& filter, const BBox& ou
     }
 }
 
+void CrayfishDataSetUtils::populateMaskFilter(DataSet& filter, const char* maskWkt) const
+{
+    filter.deleteOutputs();
+    GEOSGeometry* maskGeom;
+    initGEOS(NULL, NULL);
+    GEOSWKTReader* reader = GEOSWKTReader_create();
+
+    reader = GEOSWKTReader_create();
+    maskGeom = GEOSWKTReader_read(reader, maskWkt);
+
+    if (mOutputType == Output::TypeNode) {
+        NodeOutput* output = new NodeOutput();
+        output->init(mMesh->nodes().size(),
+                     mMesh->elements().size(),
+                     DataSet::Scalar);
+        output->time = mTimes[0];
+        for (int i = 0; i < mMesh->nodes().size(); ++i)
+        {
+            const char* pointWkt = mMesh->projectedNode(i).toWkt();
+            GEOSGeometry* pointGeom;
+            pointGeom = GEOSWKTReader_read(reader, pointWkt);
+
+            char a = GEOSIntersects(maskGeom, pointGeom);
+              if (a == 0) {
+                  output->getValues()[i] = F_FALSE;
+              } else if(a == 1) {
+                  output->getValues()[i] = F_TRUE;
+              } else {
+                  output->getValues()[i] = F_FALSE;
+              }
+        }
+        memset(output->getActive().data(), 1, mMesh->elements().size());
+        filter.addOutput(output);
+    } else {
+        ElementOutput* output = new ElementOutput();
+        output->init(mMesh->elements().size(), false);
+        output->time = mTimes[0];
+        for (int i = 0; i < mMesh->elements().size(); ++i) {
+            const char* bbox = mMesh->projectedBBox(i).toWkt();
+            GEOSGeometry* pointBBoxGeom = GEOSWKTReader_read(reader, bbox);
```

Comment:
Aren't geometries leaked here? There is API call in geos to destroy geometry objects